game/core/src/me/lihq/game/GameMain.java

```diff
@@ -164,18 +168,94 @@ public void initialiseAllPeople()
         player.setRoom(gameMap.getRoom(0));
 
+        //TODO: Add NPC assets
+        NPC npc = new NPC("Colin", "colin.png", 15, 17, gameMap.getRoom(0), true);
+        NPCs.add(npc);
+
+        NPC npc2 = new NPC("Diana", "diana.png",4,4, gameMap.getRoom(1), true);
+        NPCs.add(npc2);
+
+        NPC npc3 = new NPC("Lily", "lily.png", 0, 0, gameMap.getRoom(0), true);
+        NPCs.add(npc3);
+
+        NPC npc4 = new NPC("Mary", "mary.png", 0, 0, gameMap.getRoom(0), true);
+        NPCs.add(npc4);
+
+        NPC npc5 = new NPC("Mike", "mike.png", 0, 0, gameMap.getRoom(0), true);
+        NPCs.add(npc5);
+
+        NPC npc6 = new NPC("Will", "will.png", 0, 0, gameMap.getRoom(0), true);
+        NPCs.add(npc6);
+
+        int amountOfRooms = gameMap.getAmountOfRooms();
+
+        List<Integer> roomsLeft = new ArrayList<Integer>();
+
+        for (int i = 0; i < amountOfRooms; i ++)
         {
-            //TODO: Add NPC assets
-            NPC npc = new NPC("Bill", "player.png", 4, 4, 1, true);
+            roomsLeft.add(i);
+        }
+
+        for (NPC loopNpc : NPCs)
+        {
+            /*
+            Refill the rooms left list if there are more NPCs than Rooms. This will put AT LEAST one NPC per room if so.
+             */
+            if (roomsLeft.isEmpty())
+            {
+                for (int i = 0; i < amountOfRooms; i ++)
+                {
+                    roomsLeft.add(i);
+                }
+            }
+
+            /*
+            Pick a random room and put that NPC in it
+             */
+            int toTake = new Random().nextInt(roomsLeft.size() - 1);
+            int selectedRoom = roomsLeft.get(toTake);
+            roomsLeft.remove(toTake);
 
-            NPCs.add(npc);
+            loopNpc.setRoom(gameMap.getRoom(selectedRoom));
+            Vector2Int position = getRandomLocation(gameMap.getRoom(selectedRoom));
+            loopNpc.setTileCoordinates(position.x, position.y);
+
+            System.out.println(loopNpc.getName() + " has been placed in room " + selectedRoom + " at " + position);
         }
+    }
+
+    public Vector2Int getRandomLocation(Room room)
+    {
+        int roomWidth = ((TiledMapTileLayer) room.getTiledMap().getLayers().get(0)).getWidth();
+        int roomHeight = ((TiledMapTileLayer) room.getTiledMap().getLayers().get(0)).getHeight();
 
+        List<Vector2Int> possibleLocations = new ArrayList<Vector2Int>();
+
+        for (int w = 0; w < roomWidth; w ++)
         {
-            NPC npc = new NPC("Bob", "player.png",4,4, 2, true);
+            for (int h = 0; h < roomHeight; h ++)
+            {
+                if (room.isWalkableTile(w, h))
+                {
+                    possibleLocations.add(new Vector2Int(w, h));
+                }
+            }
+        }
 
+        Collections.shuffle(possibleLocations);
+
+        return possibleLocations.get(0);
+    }
 
-            NPCs.add(npc);
+    public List<? extends Sprite> getNPCS(Room room)
```

Comment:
doc string (I know thats my fault, just putting it as a reminder for one of us to sort)